keras_nlp/models/llama/dynamic_ntk_scaled_rotary_embedding.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2023 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from keras_nlp.backend import ops
+from keras_nlp.layers.modeling.rotary_embedding import RotaryEmbedding
+
+
+class DynamicNTKScaledRotaryEmbedding(RotaryEmbedding):
+    def _get_inverse_freq(self, rotary_dim, sequence_length):
+        freq_range = ops.arange(0, rotary_dim, 2, dtype="float32")
+        freq_range = ops.cast(freq_range, self.compute_dtype)
+        freq_range = freq_range / ops.cast(
+            self.scaling_factor, self.compute_dtype
+        )
+        if sequence_length > self.max_sequence_length:
```

Comment:
I don't think that `self.max_sequence_length` is even set? This whole layer seems probably untested?

Reply:
yes, it is getting set.